core/src/main/java/org/elasticsearch/action/bulk/TransportShardBulkAction.java

```diff
@@ -414,13 +414,6 @@ private static BulkItemResultHolder executeUpdateRequest(UpdateRequest updateReq
         FAILURE
     }
 
-    static {
-        assert Version.CURRENT.minimumCompatibilityVersion().after(Version.V_5_0_0) == false:
```

Comment:
Do you want to remove the stuff in the TODO below like the note asks? I don't think we're likely remember to remove it unless you do it now or open an issue to do it as a followup to this PR.